sklearn/metrics/_plot/confusion_matrix.py

```diff
@@ -83,22 +83,38 @@ def plot(self, include_values=True, cmap='viridis',
         n_classes = cm.shape[0]
         self.im_ = ax.imshow(cm, interpolation='nearest', cmap=cmap)
         self.text_ = None
+        if (values_format is None and cm.dtype.kind == "f"):
+            values_format = '.2g'
```

Comment:
merge this condition with the rest below? It's better to have all related logic at the same place